### PR TITLE
Add display of loss and accuracy to examples/train.py

### PR DESCRIPTION
**Pull request by akiyo0**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Let beginners better understand how to output the items they want on the console

## Modification
in model.train_step of example added log_vars and num_samples to output dict

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: examples/train.py [hunk 1]
```diff
@@ -53,25 +55,26 @@ def train_step(self, data, optimizer):
     ])
     trainset = CIFAR10(
         root='data', train=True, download=True, transform=transform)
+
     trainloader = DataLoader(
         trainset, batch_size=128, shuffle=True, num_workers=2)
 
-    optimizer = optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
```
Comment (HAOCHENYE) [inline examples/train.py:59]: Why should we modify these parameters? Such as type of optimize, max_epochs, interval, .etc.

**Review by HAOCHENYE — COMMENTED** [review]
Thanks for your contribution! It could be better to show the log in terminal in PR message.

File: examples/train.py (hunk 1 above)
Comment (carlylezqy) [reply examples/train.py:59]: Using adam can get faster convergence in this case. That is an optional modification.

**Review by HAOCHENYE — APPROVED** [review]
LGTM.

**Comment by HAOCHENYE** [comment]
Hi. Please sign the CLA :smile:

**Comment by carlylezqy** [comment]
@HAOCHENYE Hi, thanks for your reminder.
I have signed the CLA, but it still shows 'Some checks were not successful', What should I do?

**Comment by HAOCHENYE** [comment]
The failure of circle CI is not related to this PR, and we'll merge this PR ASAP!.

**Comment by OpenMMLab-Assistant001** [comment]
Hi @carlylezqy  ！We are grateful for your efforts in helping improve this open-source project during your personal time.

Welcome to join OpenMMLab Special Interest Group (SIG) private channel on **Discord**, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "**mmsig + Github ID**" as a remark when adding friends：）

Thank you again for your contribution❤
